orderer/consensus/etcdraft/util.go

```diff
@@ -84,22 +86,32 @@ func MetadataHasDuplication(md *etcdraft.ConfigMetadata) error {
 }
 
 // MetadataFromConfigValue reads and translates configuration updates from config value into raft metadata
-func MetadataFromConfigValue(configValue *common.ConfigValue) (*etcdraft.ConfigMetadata, error) {
+func MetadataFromConfigValue(configValue *common.ConfigValue) (*etcdraft.ConfigMetadata, *orderer.ConsensusType, error) {
 	consensusTypeValue := &orderer.ConsensusType{}
 	if err := proto.Unmarshal(configValue.Value, consensusTypeValue); err != nil {
-		return nil, errors.Wrap(err, "failed to unmarshal consensusType config update")
+		return nil, nil, errors.Wrap(err, "failed to unmarshal consensusType config update")
+	}
+
+	if consensusTypeValue.Type != "etcdraft" {
+		if consensusTypeValue.Type == "BFT" {
+			updatedBftMetadata := &smartbft.Options{}
+			if err := proto.Unmarshal(consensusTypeValue.Metadata, updatedBftMetadata); err != nil {
+				return nil, nil, errors.Wrap(err, "failed to unmarshal updated (new) bft metadata configuration")
+			}
```

Comment:
actually, I think lets not import smartbft package here... don't unmarshal to &smartbft.Options{}